recruit_flow_ai/ai.py

```diff
@@ -0,0 +1,218 @@
+"""
+This module contains the RecruitFlowAI class, which is an AI assistant for the 
+RecruitFlow application.
+
+The RecruitFlowAI class provides methods for generating responses based on given 
+messages using the OpenAI API.
+
+Example:
+    # Create an instance of RecruitFlowAI
+    ai = RecruitFlowAI(api_key="YOUR_API_KEY")
+
+    # Generate a response
+    response = ai.generate_response(openai_msgs=[
+        {"role": "system", "content": "You are a recruitment assistant."},
+        {"role": "user", "content": "What questions should I ask during the screening process"}
+    ])
+
+    # Print the response
+    logger.error(response)
+
+"""
+
+import openai
+
+import logging
+import re   # used for api key format validation
+import os   # used to read OPANAI_API_KEY env variable
+from dotenv import load_dotenv
+from pathlib import Path
+
+from .parse_config import parse_config, ConfigModel
+
+PROMPTS_CONFIG_PATH = "recruit_flow_ai/prompts_config.json"
+ISSUES_REPORT_MSG = "Report this to #recruitflowai_issues channel."
+
+logFormatter = logging.Formatter("%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
```

Comment:
I think we can config here root logger with basicConfig
https://docs.python.org/3/library/logging.html#logging.basicConfig

```
logging.basicConfig(level=logging.DEBUG, format='"%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
logger = logging.getLogger(__name__)
```
Why do we need thread name in the logs? 
I would add logger name instead of it `%(name)s`

Reply:
let's ignore logging in this PR, it will be refactored later in a separate issue. I want to have it configurable for all files from the one file, although I'm not sure yet how to do it